// @pulumi/github/repository.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Repository = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * This resource allows you to create and manage repositories within your
 * GitHub organization or personal account.
 *
 * > **Note** When used with GitHub App authentication, even GET requests must have the `contents:write` permission. Without it, the following arguments will be ignored, leading to unexpected behavior and confusing diffs: `allowMergeCommit`, `allowSquashMerge`, `allowRebaseMerge`, `mergeCommitTitle`, `mergeCommitMessage`, `squashMergeCommitTitle` and `squashMergeCommitMessage`.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as github from "@pulumi/github";
 *
 * const example = new github.Repository("example", {
 *     name: "example",
 *     description: "My awesome codebase",
 *     visibility: "public",
 *     template: {
 *         owner: "github",
 *         repository: "terraform-template-module",
 *         includeAllBranches: true,
 *     },
 * });
 * ```
 *
 * ### With GitHub Pages Enabled
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as github from "@pulumi/github";
 *
 * const example = new github.Repository("example", {
 *     name: "example",
 *     description: "My awesome web page",
 *     "private": false,
 *     pages: {
 *         source: {
 *             branch: "master",
 *             path: "/docs",
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Repositories can be imported using the `name`, e.g.
 *
 * ```sh
 * $ pulumi import github:index/repository:Repository terraform terraform
 * ```
 */
class Repository extends pulumi.CustomResource {
    /**
     * Get an existing Repository resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Repository(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Repository.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Repository.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["allowAutoMerge"] = state?.allowAutoMerge;
            resourceInputs["allowMergeCommit"] = state?.allowMergeCommit;
            resourceInputs["allowRebaseMerge"] = state?.allowRebaseMerge;
            resourceInputs["allowSquashMerge"] = state?.allowSquashMerge;
            resourceInputs["allowUpdateBranch"] = state?.allowUpdateBranch;
            resourceInputs["archiveOnDestroy"] = state?.archiveOnDestroy;
            resourceInputs["archived"] = state?.archived;
            resourceInputs["autoInit"] = state?.autoInit;
            resourceInputs["defaultBranch"] = state?.defaultBranch;
            resourceInputs["deleteBranchOnMerge"] = state?.deleteBranchOnMerge;
            resourceInputs["description"] = state?.description;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["fullName"] = state?.fullName;
            resourceInputs["gitCloneUrl"] = state?.gitCloneUrl;
            resourceInputs["gitignoreTemplate"] = state?.gitignoreTemplate;
            resourceInputs["hasDiscussions"] = state?.hasDiscussions;
            resourceInputs["hasDownloads"] = state?.hasDownloads;
            resourceInputs["hasIssues"] = state?.hasIssues;
            resourceInputs["hasProjects"] = state?.hasProjects;
            resourceInputs["hasWiki"] = state?.hasWiki;
            resourceInputs["homepageUrl"] = state?.homepageUrl;
            resourceInputs["htmlUrl"] = state?.htmlUrl;
            resourceInputs["httpCloneUrl"] = state?.httpCloneUrl;
            resourceInputs["ignoreVulnerabilityAlertsDuringRead"] = state?.ignoreVulnerabilityAlertsDuringRead;
            resourceInputs["isTemplate"] = state?.isTemplate;
            resourceInputs["licenseTemplate"] = state?.licenseTemplate;
            resourceInputs["mergeCommitMessage"] = state?.mergeCommitMessage;
            resourceInputs["mergeCommitTitle"] = state?.mergeCommitTitle;
            resourceInputs["name"] = state?.name;
            resourceInputs["nodeId"] = state?.nodeId;
            resourceInputs["pages"] = state?.pages;
            resourceInputs["primaryLanguage"] = state?.primaryLanguage;
            resourceInputs["private"] = state?.private;
            resourceInputs["repoId"] = state?.repoId;
            resourceInputs["securityAndAnalysis"] = state?.securityAndAnalysis;
            resourceInputs["squashMergeCommitMessage"] = state?.squashMergeCommitMessage;
            resourceInputs["squashMergeCommitTitle"] = state?.squashMergeCommitTitle;
            resourceInputs["sshCloneUrl"] = state?.sshCloneUrl;
            resourceInputs["svnUrl"] = state?.svnUrl;
            resourceInputs["template"] = state?.template;
            resourceInputs["topics"] = state?.topics;
            resourceInputs["visibility"] = state?.visibility;
            resourceInputs["vulnerabilityAlerts"] = state?.vulnerabilityAlerts;
            resourceInputs["webCommitSignoffRequired"] = state?.webCommitSignoffRequired;
        }
        else {
            const args = argsOrState;
            resourceInputs["allowAutoMerge"] = args?.allowAutoMerge;
            resourceInputs["allowMergeCommit"] = args?.allowMergeCommit;
            resourceInputs["allowRebaseMerge"] = args?.allowRebaseMerge;
            resourceInputs["allowSquashMerge"] = args?.allowSquashMerge;
            resourceInputs["allowUpdateBranch"] = args?.allowUpdateBranch;
            resourceInputs["archiveOnDestroy"] = args?.archiveOnDestroy;
            resourceInputs["archived"] = args?.archived;
            resourceInputs["autoInit"] = args?.autoInit;
            resourceInputs["defaultBranch"] = args?.defaultBranch;
            resourceInputs["deleteBranchOnMerge"] = args?.deleteBranchOnMerge;
            resourceInputs["description"] = args?.description;
            resourceInputs["gitignoreTemplate"] = args?.gitignoreTemplate;
            resourceInputs["hasDiscussions"] = args?.hasDiscussions;
            resourceInputs["hasDownloads"] = args?.hasDownloads;
            resourceInputs["hasIssues"] = args?.hasIssues;
            resourceInputs["hasProjects"] = args?.hasProjects;
            resourceInputs["hasWiki"] = args?.hasWiki;
            resourceInputs["homepageUrl"] = args?.homepageUrl;
            resourceInputs["ignoreVulnerabilityAlertsDuringRead"] = args?.ignoreVulnerabilityAlertsDuringRead;
            resourceInputs["isTemplate"] = args?.isTemplate;
            resourceInputs["licenseTemplate"] = args?.licenseTemplate;
            resourceInputs["mergeCommitMessage"] = args?.mergeCommitMessage;
            resourceInputs["mergeCommitTitle"] = args?.mergeCommitTitle;
            resourceInputs["name"] = args?.name;
            resourceInputs["pages"] = args?.pages;
            resourceInputs["private"] = args?.private;
            resourceInputs["securityAndAnalysis"] = args?.securityAndAnalysis;
            resourceInputs["squashMergeCommitMessage"] = args?.squashMergeCommitMessage;
            resourceInputs["squashMergeCommitTitle"] = args?.squashMergeCommitTitle;
            resourceInputs["template"] = args?.template;
            resourceInputs["topics"] = args?.topics;
            resourceInputs["visibility"] = args?.visibility;
            resourceInputs["vulnerabilityAlerts"] = args?.vulnerabilityAlerts;
            resourceInputs["webCommitSignoffRequired"] = args?.webCommitSignoffRequired;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fullName"] = undefined /*out*/;
            resourceInputs["gitCloneUrl"] = undefined /*out*/;
            resourceInputs["htmlUrl"] = undefined /*out*/;
            resourceInputs["httpCloneUrl"] = undefined /*out*/;
            resourceInputs["nodeId"] = undefined /*out*/;
            resourceInputs["primaryLanguage"] = undefined /*out*/;
            resourceInputs["repoId"] = undefined /*out*/;
            resourceInputs["sshCloneUrl"] = undefined /*out*/;
            resourceInputs["svnUrl"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Repository.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Repository = Repository;
/** @internal */
Repository.__pulumiType = 'github:index/repository:Repository';
