// @pulumi/github/repositoryPullRequest.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryPullRequest = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * This resource allows you to create and manage PullRequests for repositories within your GitHub organization or personal account.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as github from "@pulumi/github";
 *
 * const example = new github.RepositoryPullRequest("example", {
 *     baseRepository: "example-repository",
 *     baseRef: "main",
 *     headRef: "feature-branch",
 *     title: "My newest feature",
 *     body: "This will change everything",
 * });
 * ```
 */
class RepositoryPullRequest extends pulumi.CustomResource {
    /**
     * Get an existing RepositoryPullRequest resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RepositoryPullRequest(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of RepositoryPullRequest.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RepositoryPullRequest.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["baseRef"] = state?.baseRef;
            resourceInputs["baseRepository"] = state?.baseRepository;
            resourceInputs["baseSha"] = state?.baseSha;
            resourceInputs["body"] = state?.body;
            resourceInputs["draft"] = state?.draft;
            resourceInputs["headRef"] = state?.headRef;
            resourceInputs["headSha"] = state?.headSha;
            resourceInputs["labels"] = state?.labels;
            resourceInputs["maintainerCanModify"] = state?.maintainerCanModify;
            resourceInputs["number"] = state?.number;
            resourceInputs["openedAt"] = state?.openedAt;
            resourceInputs["openedBy"] = state?.openedBy;
            resourceInputs["owner"] = state?.owner;
            resourceInputs["state"] = state?.state;
            resourceInputs["title"] = state?.title;
            resourceInputs["updatedAt"] = state?.updatedAt;
        }
        else {
            const args = argsOrState;
            if (args?.baseRef === undefined && !opts.urn) {
                throw new Error("Missing required property 'baseRef'");
            }
            if (args?.baseRepository === undefined && !opts.urn) {
                throw new Error("Missing required property 'baseRepository'");
            }
            if (args?.headRef === undefined && !opts.urn) {
                throw new Error("Missing required property 'headRef'");
            }
            if (args?.title === undefined && !opts.urn) {
                throw new Error("Missing required property 'title'");
            }
            resourceInputs["baseRef"] = args?.baseRef;
            resourceInputs["baseRepository"] = args?.baseRepository;
            resourceInputs["body"] = args?.body;
            resourceInputs["headRef"] = args?.headRef;
            resourceInputs["maintainerCanModify"] = args?.maintainerCanModify;
            resourceInputs["owner"] = args?.owner;
            resourceInputs["title"] = args?.title;
            resourceInputs["baseSha"] = undefined /*out*/;
            resourceInputs["draft"] = undefined /*out*/;
            resourceInputs["headSha"] = undefined /*out*/;
            resourceInputs["labels"] = undefined /*out*/;
            resourceInputs["number"] = undefined /*out*/;
            resourceInputs["openedAt"] = undefined /*out*/;
            resourceInputs["openedBy"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RepositoryPullRequest.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RepositoryPullRequest = RepositoryPullRequest;
/** @internal */
RepositoryPullRequest.__pulumiType = 'github:index/repositoryPullRequest:RepositoryPullRequest';
