// @pulumi/github/repositoryFile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryFile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * This resource allows you to create and manage files within a
 * GitHub repository.
 *
 * ## Example Usage
 *
 * ### Existing Branch
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as github from "@pulumi/github";
 *
 * const foo = new github.Repository("foo", {
 *     name: "tf-acc-test-%s",
 *     autoInit: true,
 * });
 * const fooRepositoryFile = new github.RepositoryFile("foo", {
 *     repository: foo.name,
 *     branch: "main",
 *     file: ".gitignore",
 *     content: "**&#47;*.tfstate",
 *     commitMessage: "Managed by Pulumi",
 *     commitAuthor: "Terraform User",
 *     commitEmail: "terraform@example.com",
 *     overwriteOnCreate: true,
 * });
 * ```
 *
 * ### Auto Created Branch
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as github from "@pulumi/github";
 *
 * const foo = new github.Repository("foo", {
 *     name: "tf-acc-test-%s",
 *     autoInit: true,
 * });
 * const fooRepositoryFile = new github.RepositoryFile("foo", {
 *     repository: foo.name,
 *     branch: "does/not/exist",
 *     file: ".gitignore",
 *     content: "**&#47;*.tfstate",
 *     commitMessage: "Managed by Pulumi",
 *     commitAuthor: "Terraform User",
 *     commitEmail: "terraform@example.com",
 *     overwriteOnCreate: true,
 *     autocreateBranch: true,
 * });
 * ```
 *
 * ## Import
 *
 * Repository files can be imported using a combination of the `repo` and `file`, e.g.
 *
 * ```sh
 * $ pulumi import github:index/repositoryFile:RepositoryFile gitignore example/.gitignore
 * ```
 * To import a file from a branch other than the default branch, append `:` and the branch name, e.g.
 *
 * ```sh
 * $ pulumi import github:index/repositoryFile:RepositoryFile gitignore example/.gitignore:dev
 * ```
 */
class RepositoryFile extends pulumi.CustomResource {
    /**
     * Get an existing RepositoryFile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RepositoryFile(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of RepositoryFile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RepositoryFile.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["autocreateBranch"] = state?.autocreateBranch;
            resourceInputs["autocreateBranchSourceBranch"] = state?.autocreateBranchSourceBranch;
            resourceInputs["autocreateBranchSourceSha"] = state?.autocreateBranchSourceSha;
            resourceInputs["branch"] = state?.branch;
            resourceInputs["commitAuthor"] = state?.commitAuthor;
            resourceInputs["commitEmail"] = state?.commitEmail;
            resourceInputs["commitMessage"] = state?.commitMessage;
            resourceInputs["commitSha"] = state?.commitSha;
            resourceInputs["content"] = state?.content;
            resourceInputs["file"] = state?.file;
            resourceInputs["overwriteOnCreate"] = state?.overwriteOnCreate;
            resourceInputs["ref"] = state?.ref;
            resourceInputs["repository"] = state?.repository;
            resourceInputs["sha"] = state?.sha;
        }
        else {
            const args = argsOrState;
            if (args?.content === undefined && !opts.urn) {
                throw new Error("Missing required property 'content'");
            }
            if (args?.file === undefined && !opts.urn) {
                throw new Error("Missing required property 'file'");
            }
            if (args?.repository === undefined && !opts.urn) {
                throw new Error("Missing required property 'repository'");
            }
            resourceInputs["autocreateBranch"] = args?.autocreateBranch;
            resourceInputs["autocreateBranchSourceBranch"] = args?.autocreateBranchSourceBranch;
            resourceInputs["autocreateBranchSourceSha"] = args?.autocreateBranchSourceSha;
            resourceInputs["branch"] = args?.branch;
            resourceInputs["commitAuthor"] = args?.commitAuthor;
            resourceInputs["commitEmail"] = args?.commitEmail;
            resourceInputs["commitMessage"] = args?.commitMessage;
            resourceInputs["content"] = args?.content;
            resourceInputs["file"] = args?.file;
            resourceInputs["overwriteOnCreate"] = args?.overwriteOnCreate;
            resourceInputs["repository"] = args?.repository;
            resourceInputs["commitSha"] = undefined /*out*/;
            resourceInputs["ref"] = undefined /*out*/;
            resourceInputs["sha"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RepositoryFile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RepositoryFile = RepositoryFile;
/** @internal */
RepositoryFile.__pulumiType = 'github:index/repositoryFile:RepositoryFile';
